src/v2i-hub/CDASimAdapter/src/CDASimConnection.cpp

```diff
@@ -217,4 +224,67 @@ namespace CDASimAdapter{
         forward_message( msg , message_receiver_publisher );
     }
 
+    void CDASimConnection::populate_sensors_with_file(const std::string file_path, Json::Value& sensors_json_v){
```

Comment:
code smell with this method signature. Change to `const std::string &file_path`

Reply:
Also not a fan of this being a void function that requires a JSON value that is populated and returned by reference. Is there a reason you opted for this instead of just returning a JSON value.

Reply:
Addressed

Reply:
This is to reuse/populate the global Json value already defined in the plugin. As the global variable is already there, I do not want to return a copy of this json::value and assign it to the global variable again.

Reply:
I will let you decide this, my argument will be that this is a one time start-up operation where I think readability is more important that efficiency. I do not think there is significant difference in a one time operation like this.

Reply:
Removed the return

Reply:
Refactored the func